Common/Nightmare/MasochistNPC.cs

```diff
@@ -100,45 +104,57 @@ internal class MasochistNPC : GlobalNPC {
 				spawnRate -= 10;
 			}
 		}
+		public override bool PreAI(NPC npc) {
+			return base.PreAI(npc);
+		}
 		public override void AI(NPC npc) {
 			if (ModContent.GetInstance<BossRushModConfig>().Nightmare) {
-				if (npc.type == NPCID.KingSlime) {
-					KingSlimeAI(npc);
-				}
-				return;
+				//if (npc.aiStyle == 15) {
+				//	KingSlimeAI(npc);
+				//}
+				//return;
 			}
 			base.AI(npc);
 		}
+		/// <summary>
+		/// We need to deeply interfere with KS ai and so I port entire KS code from vanilla<br/>
+		/// We could have done it like fargo but I don't like that approach cause it is too limiting
+		/// </summary>
+		/// <param name="npc"></param>
 		private void KingSlimeAI(NPC npc) {
 			float num236 = 1f;
 			float num237 = 1f;
-			bool flag6 = false;
 			bool flag7 = false;
 			bool flag8 = false;
```

Comment:
Please name variables something sensible instead of `num236`, `flag7`, etch.

Reply:
ah that vanilla boss code ( complier artifact ), I'm still trying to port them over to our mod